builtin/game/misc.lua

```diff
@@ -99,7 +101,9 @@ core.register_on_leaveplayer(function(player, timed_out)
 	if timed_out then
 		announcement = announcement .. " (timed out)"
 	end
-	core.chat_send_all(announcement)
+	if not core.is_singleplayer() then
```

Comment:
This conditional should be above "local announcement = " for optimization and to make it clear the announcement isn't being used for anything at all in singleplayer. :+1: then.